internal/bsr/bsr.go

```diff
@@ -215,10 +216,22 @@ func OpenSession(ctx context.Context, sessionRecordingId string, f storage.FS, k
 		return nil, err
 	}
 
+	af, ok := SummaryAllocFuncs.Get(meta.Protocol, SessionContainer)
+	if !ok {
+		return nil, fmt.Errorf("%s: failed to get summary type", op)
```

Comment:
Do we have any unhappy path unit testing to validate this check?

Reply:
I just added unit tests to cover this:
https://github.com/hashicorp/boundary/blob/d668531c764f7d2f59d30c5a13786c338a8ecbc1/internal/bsr/bsr_open_test.go#L331